Common/src/linear_algebra/CSysSolve.cpp

```diff
@@ -923,6 +923,7 @@ unsigned long CSysSolve<ScalarType>::Solve(CSysMatrix<ScalarType> & Jacobian, co
     Iterations = IterLinSol;
   }
   END_SU2_OMP_MASTER
+  SU2_OMP_BARRIER
```

Comment:
I don't think this is strictly necessary (for Residual and Iterations at least) because it is also the master thread that will request those fields to store the results in CSolver after the linear system is solved. Did you run into some issues?

Reply:
The *suspected* bug was the following: If the master thread is significantly ahead, the shared variables `LinSysRes_ptr` and `LinSysSol_ptr` are reset in `HandleTemporariesOut` although other threads still need them. The code looks like a barrier to prevent this is missing, but after rechecking I think the one in `CSysVector::PassiveCopy` called from `HandleTemporariesOut` does the trick, and the corner case of `(const void*)this == (const void*)&other` that skips this barrier is irrelevant for the part of code in question?

Reply:
resolved in fcc39ce28bd66586a9854e97af43a65fdb6d2c82 as discussed in the dev meeting